### DISCUSSION TITLE
catalogue.RegistryError: [E893] Could not find function 'spacy.copy_from_base_model.v1' in function registry 'callbacks'

### DISCUSSION THREAD

**Discussion by ltadjou** [explosion/spaCy#7985] [Help: Coding & Implementations] [answered]
I'm Spacy's new user and I'm trying to run this [ner_demo_update](https://github.com/explosion/projects/tree/v3/pipelines/ner_demo_update) project and I got this error : 
_catalogue.RegistryError: [E893] Could not find function 'spacy.copy_from_base_model.v1' in function registry 'callbacks'. If you're using a custom function, make sure the code is available. If the function is provided by a third-party package, e.g. spacy-transformers, make sure the package is installed in your environment._
I would like to know if anyone faced the same problem

**Comment by adrianeboyd** [comment, marked as answer]
Hi, please update to spacy v3.0.6+.

I think that `spacy project` commands should show a warning if you have an earlier version than this, but I can double-check that all the settings in this demo project are correct...

**Reply by adrianeboyd** [reply]
Ah, it shows up as an error. I see this if I try to run `spacy project run all` with spaCy v3.0.5:

```text
✘ The project.yml specifies a spaCy version range (>=3.0.6,<4.0.0)
that's not compatible with the version of spaCy you're running (3.0.5). You can
edit version requirement in the project.yml to load it, but the project may not
run as expected.
```

**Reply by ltadjou** [reply]
Thanks @adrianeboyd , the problem was  the Spacy's version, I was on 3.0.5 and I upgraded to 3.0.6 . 

All is OK Thanks

**Reply by mdrazak2001** [reply]
My version is 3.2.1 still i am getting this error

**Reply by polm** [reply]
@mdrazak2001 Can you share the output of `spacy info`? Versions 3.0.6 or higher should contain this without any external packages so no idea how that could happen...

**Reply by dhruvbhimani-2512** [reply]
@polm @adrianeboyd I need to train my model in spaCy v3.0.5 and I am getting this error. How can I train it in this version without updating it to v3.0.6+?

**Reply by polm** [reply]
This particular project is not intended to run with <3.0.6, so you'll need to upgrade. If you are not running this particular project, please provide more details.

Can you clarify why you can't upgrade? Changes between 3.0.5 and 3.0.6 shouldn't be that drastic, and in particular it shouldn't break anything.

**Reply by dhruvbhimani-2512** [reply]
Hey @polm , thanks for your reply. 

I am building custom NER component in a blank model. I am using v3.0.5 because my organisation recommended it. I have already trained the model in v3.0.6 and it is working absolutely fine. But I am getting the same error mentioned here in 3.0.5. How to solve it or is there any different method to train for older versions?

**Reply by polm** [reply]
We don't have any particular alternative to upgrading in this case, no. You could copy the relevant commits into your local code but that's basically a more complicated way of upgrading. I would suggest you discuss upgrading with your organization.

**Reply by dhruvbhimani-2512** [reply]
> 

Then how models were being trained before v3.0.6? Could you please tell?

**Reply by polm** [reply]
You can use `spacy init config` to create a config compatible with the particular version you're using.
